CHANGES.md

```diff
@@ -56,13 +56,50 @@
   
   the load order is:
 
-  - ./config/twitcher
   - ./config/magpie (loaded as a dependency of twitcher, not loaded a second time after project-api)
-  - ./config/project-api 
-  - ./optional-components/generic_bird
+  - ./config/twitcher
+  - ./config/project-api
   - ./config/wps_outputs-volume (loaded as a dependency of generic_bird)
+  - ./optional-components/generic_bird
   - ./components/cowbird
 
+  This load order also applies to the order that docker-compose-extra.yml files are specified. If a component also
+  includes an override file for another component (eg: ./config/finch/config/proxy/docker-compose-extra.yml overrides 
+  ./config/proxy/docker-compose-extra.yml), the following additional load order rules apply:
+
+  - if the component that is being overridden has already been loaded, the override file is loaded immediately
+  - otherwise, the override files will be loaded immediately after the component that is being overridden has been loaded
+
+  For example, with the following files in place:
+
+    ```shell
+  # env.local
+  DEFAULT_CONF_DIRS="
+    ./config/finch
+    ./config/proxy
+  "
+  ```
+  ```yaml
+  # config/proxy/docker-compose-extra.yml
+    ...
+  # config/finch/docker-compose-extra.yml
+    ...
+  # config/finch/config/proxy/docker-compose-extra.yml
+    ...
+  ```
+
+  the docker compose files will be loaded in the following order: 
+
+  - config/finch/docker-compose-extra.yml
+  - config/proxy/docker-compose-extra.yml
+  - config/finch/config/proxy/docker-compose-extra.yml
```

Comment:
This seems to contradict what the test evaluates, or there is something I missed? 
I'm expecting `proxy` to be first as `finch` depends on it to provide its extra compose YAML.

Reply:
Finch doesn't have proxy as an explicit dependency (it is not included in the COMPONENT_DEPENDENCIES variable defined in config/finch/default.env) so proxy is not loaded before finch. Because of this, the order that the components appear in the DEFAULT_CONF_DIRS list is respected (finch is before proxy). However, the file that overrides proxy settings (config/finch/config/proxy/docker-compose-extra.yml) can't be loaded until _after_ both finch and proxy are loaded, since it could potentially override settings in either one, so it is loaded as soon as both of finch an proxy are loaded. 

From the CHANGES.md file, here is the relevant explanation:

>   This load order also applies to the order that docker-compose-extra.yml files are specified. If a component also
  includes an override file for another component (eg: ./config/finch/config/proxy/docker-compose-extra.yml overrides 
  ./config/proxy/docker-compose-extra.yml), the following additional load order rules apply:
>
>  - if the component that is being overridden has already been loaded, the override file is loaded immediately
>  - otherwise, the override files will be loaded immediately after the component that is being overridden has been loaded

Reply:
I see. So in the tests, `proxy` is loaded before `finch` because of some other component that defined it needed `proxy` as dependency, but considering only `proxy`/`finch` by themselves in this example, they have no particular order. Only the extra should be last, as it needs both.